### PR TITLE
maintain DNS server assignment on tun interface

### PR DESCRIPTION
**Pull request by ekoby**
this change addresses the issue of DNS server removed from tun interface on any network change event

### CODE REVIEW

File: programs/ziti-edge-tunnel/netif_driver/linux/tun.c [hunk 1]
```diff
@@ -24,8 +29,25 @@
...
 
+static struct {
+    char tun_name[16];
+    uint32_t dns_ip;
+
+    uv_udp_t nl_udp;
+    uv_timer_t update_timer;
+} dns_maintainer;
+
+typedef struct nl_req_s nl_req_t;
```
Comment (scareything) [inline programs/ziti-edge-tunnel/netif_driver/linux/tun.c:44]: Is there a reason you didn't combine the typedef with the struct definition?

File: programs/ziti-edge-tunnel/netif_driver/linux/tun.c [hunk 2]
```diff
@@ -24,8 +29,25 @@
 #define IP_BIN "/sbin/ip "
 #endif
 
+#define CHECK_UV(op) do{ int rc = op; if (rc < 0) ZITI_LOG(ERROR, "uv_err: %d/%s", rc, uv_strerror(rc)); } while(0)
+
 #define RESOLVECTL "resolvectl"
 
+static struct {
+    char tun_name[16];
```
Comment (scareything) [inline programs/ziti-edge-tunnel/netif_driver/linux/tun.c:37]: This would be a good place to use `IFNAMSIZ`

**Review by scareything — APPROVED** [review]

File: programs/ziti-edge-tunnel/netif_driver/linux/tun.c (hunk 2 above)
Comment (ekoby) [reply programs/ziti-edge-tunnel/netif_driver/linux/tun.c:37]: :+1:

File: programs/ziti-edge-tunnel/netif_driver/linux/tun.c (hunk 1 above)
Comment (ekoby) [reply programs/ziti-edge-tunnel/netif_driver/linux/tun.c:44]: this is not needed (yet?), cleaning it up

**Pull request merged**